### PR TITLE
Search add-on with Flow UI (1722)

### PR DESCRIPTION
**Pull request by breakingfox**

### CODE REVIEW

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/utils/FullTextFilterUtils.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.searchflowui.utils;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
```
Comment (NikitaShchienko) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/utils/FullTextFilterUtils.java:20]: Add Java docs

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/view/result/SearchResultsView.java [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    protected EntitySearcher entitySearcher;
+    @Autowired
+    protected SearchStrategyManager searchStrategyManager;
+
+    @ViewComponent
+    protected VerticalLayout contentBoxWithName;
+
+    protected SearchResult searchResult;
+
+    protected SearchStrategy searchStrategy;
```
Comment (NikitaShchienko) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/view/result/SearchResultsView.java:90]: Format the code.

1 - Constants
2 - ViewComponent
3 - Autowired
4 - Variables

Example - https://github.com/jmix-framework/jmix/blob/master/jmix-quartz/quartz-flowui/src/main/java/io/jmix/quartzflowui/view/trigger/TriggerModelDetailView.java

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import io.jmix.search.searching.SearchStrategy;
+import io.jmix.searchflowui.view.result.SearchResultsView;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.jmix.searchflowui.view.result.SearchResultsView.*;
+
+//todo add studio component annotation
```
Comment (glebfox) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java:41]: do you need this comment?

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        searchProperties = applicationContext.getBean(SearchProperties.class);
+        viewNavigators = applicationContext.getBean(ViewNavigators.class);
+        dialogWindows = applicationContext.getBean(DialogWindows.class);
+    }
+
+    @Override
+    protected void initComponent() {
+        super.initComponent();
+
+        getValue();
```
Comment (glebfox) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java:81]: ?

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        initSearchField();
+
+        addValueChangeListener(valueChangeEvent -> {
```
Comment (glebfox) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java:84]: ?

File: jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+import io.jmix.searchflowui.view.result.SearchResultsView;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.jmix.searchflowui.view.result.SearchResultsView.*;
+
+//todo add studio component annotation
+public class SearchField extends TypedTextField<String> {
```
Comment (glebfox) [inline jmix-search/search-flowui/src/main/java/io/jmix/searchflowui/component/SearchField.java:42]: I doubt you need all the APIs from TextField. It seems that `CustomField` is more suitable base class

**Review by glebfox — APPROVED** [review]

**Pull request merged**